src/main/java/hudson/plugins/sonar/SonarGlobalConfiguration.java

```diff
@@ -0,0 +1,127 @@
+ *
+ * Jenkins Plugin for SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package hudson.plugins.sonar;
+
+import hudson.CopyOnWrite;
+import hudson.Extension;
+import hudson.plugins.sonar.SonarPublisher.DescriptorImpl;
+import hudson.plugins.sonar.utils.Logger;
+import hudson.plugins.sonar.utils.SQServerVersions;
+import hudson.util.FormValidation;
+import hudson.util.ListBoxModel;
+import hudson.util.ListBoxModel.Option;
+import jenkins.model.GlobalConfiguration;
+import jenkins.model.Jenkins;
+import net.sf.json.JSONObject;
+import org.apache.commons.lang.ArrayUtils;
+import org.apache.commons.lang.StringUtils;
+import org.kohsuke.stapler.QueryParameter;
+import org.kohsuke.stapler.StaplerRequest;
+
+import java.util.List;
+
+@Extension(ordinal = 100)
+public class SonarGlobalConfiguration extends GlobalConfiguration {
+  @CopyOnWrite
+  private volatile SonarInstallation[] installations = new SonarInstallation[0];
+  private volatile boolean buildWrapperEnabled = false;
+  private transient boolean migrated = false;
+
+  public SonarGlobalConfiguration() {
+    load();
+    save();
+  }
+
+  /**
+   * @return all configured {@link hudson.plugins.sonar.SonarInstallation}
+   */
+  public SonarInstallation[] getInstallations() {
+    migrate();
+    return installations;
+  }
+
+  public boolean isBuildWrapperEnabled() {
+    migrate();
+    return buildWrapperEnabled;
+  }
+
+  public void setInstallations(SonarInstallation... installations) {
+    this.installations = installations;
+    save();
+  }
+
+  public void setBuildWrapperEnabled(boolean enabled) {
+    this.buildWrapperEnabled = enabled;
+    save();
+  }
+
+  /**
+   * Attempts to migrated data from SonarPublished, which was previously holding the global configuration.
+   * It is thread safe and will refuse to migrate if a SonarQube installation already exists in this class.
+   * Migration will only be attempted once. 
+   */
+  private void migrate() {
+    if (migrated) {
+      return;
+    }
+
+    synchronized (this) {
+      if (migrated) {
+        return;
+      }
+      // SonarPublisher might be null if Maven plugin is disabled or not installed
+      DescriptorImpl publisher = Jenkins.getInstance().getDescriptorByType(SonarPublisher.DescriptorImpl.class);
+      if (publisher != null && publisher.getInstallations() != null && publisher.getInstallations().length > 0) {
+
+        if (ArrayUtils.isEmpty(this.installations)) {
+          this.installations = publisher.getInstallations();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this use of "getInstallations"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)